clusterstate/provider/aws.go

```diff
@@ -0,0 +1,227 @@
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	awsclient "github.com/giantswarm/e2eclients/aws"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	EC2InstanceType = "m5.xlarge"
+)
+
+type AWSConfig struct {
+	AWSClient      *awsclient.Client
+	GuestFramework *framework.Guest
+	HostFramework  *framework.Host
+	Logger         micrologger.Logger
+
+	ClusterID string
+}
+
+type AWS struct {
+	awsClient      *awsclient.Client
+	guestFramework *framework.Guest
+	hostFramework  *framework.Host
+	logger         micrologger.Logger
+
+	clusterID string
+}
+
+func NewAWS(config AWSConfig) (*AWS, error) {
+	if config.AWSClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWSClient must not be empty", config)
+	}
+	if config.GuestFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestFramework must not be empty", config)
+	}
+	if config.HostFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostFramework must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+
+	a := &AWS{
+		awsClient:      config.AWSClient,
+		guestFramework: config.GuestFramework,
+		hostFramework:  config.HostFramework,
+		logger:         config.Logger,
+
+		clusterID: config.ClusterID,
+	}
+
+	return a, nil
+}
+
+func (a *AWS) InstallTestApp() error {
```

Comment:
Installing the test app should actually be independent from the provider because this is just APPR/CNR/K8s.